templates/login.html

```diff
@@ -0,0 +1,17 @@
+{% extends "layout.html" %}
+
+{% block title %}Login{% endblock %}
+
+{% block content %}
+  <form method="POST" class="login-form js-login-form">
+    <input type="text" name="username" placeholder="Username">
+    <button>Sign up or Log in</button>
+  </form>
+
+  <script data-username="{{ username }}">
+    if (document.currentScript.dataset.username) {
+      window.opener.location.reload();
+      window.close();
```

Comment:
This is a bit hacky, but if the user is logged in when the page loads (the Python view function passes `username` to the template if the user is logged in) then it reloads the parent window and closes the popup window. What happens is: popup window opens (user is not logged in), submitting the login form does a reload of the popup window and this time the user _is_ logged in so it does the close and reload.